### PR TITLE
Simplify compilation for other platfroms

### PR DESCRIPTION
**Pull request by Solpatium**
Please note that I am not a C++ expert, so I'll appreciate all feedback.
I included all changes necessary to get it working on my machine, but in the meantime some problems were fixed by other PRs. Starknet.py uses this code to build native code (https://github.com/software-mansion/starknet.py).

Changes:

- Added gtest as recommended in https://google.github.io/googletest/quickstart-cmake.html and added gmock to target_link_libraries for tests.
- Made src/starkware/crypto/ffi/js/CMakeLists.txt copy either .so or .dylib files.
- Added portable endian header to allow compillation of platforms that don't support endian.h based on https://github.com/sahlberg/libsmb2/blob/master/include/portable-endian.h. https://github.com/starkware-libs/crypto-cpp/pull/4 includes headers for Apple devices.
- Replaced bn with bn.js in js package to fix js tests. https://github.com/starkware-libs/crypto-cpp/pull/7 fixes it as well.
- Added uint128 substitute for 32 bit platforms using [boost](https://www.boost.org/doc/libs/1_72_0/libs/multiprecision/doc/html/boost_multiprecision/intro.html).

### CODE REVIEW

File: src/starkware/crypto/ffi/js/CMakeLists.txt [hunk 1]
```diff
@@ -1,14 +1,14 @@
 add_custom_command(
-    OUTPUT ${CMAKE_CURRENT_SOURCE_DIR}/libcrypto_c_exports.so
+    OUTPUT ${CMAKE_CURRENT_SOURCE_DIR}/libcrypto
     COMMAND
         ${CMAKE_COMMAND} -E copy
-        ${CMAKE_CURRENT_BINARY_DIR}/../libcrypto_c_exports.so
-        ${CMAKE_CURRENT_SOURCE_DIR}/libcrypto_c_exports.so
+        ${CMAKE_CURRENT_BINARY_DIR}/../libcrypto_c_exports.*
```
Comment (ilyalesokhin-starkware) [inline src/starkware/crypto/ffi/js/CMakeLists.txt:5]: Why did you make this change?

**Comment by ilyalesokhin-starkware** [comment]
Please split this into multiple PRs.
1. Portable-endian.h.
2. .js changes
3. boots 128bit.
4. clang changes.

File: src/starkware/crypto/ffi/js/CMakeLists.txt (hunk 1 above)
Comment (Solpatium) [reply src/starkware/crypto/ffi/js/CMakeLists.txt:5]: To copy both .so and .dylib (mac) shared libs

**Comment by Solpatium** [comment]
Done:
1. endian.h - https://github.com/starkware-libs/crypto-cpp/pull/12
2. js deps - https://github.com/starkware-libs/crypto-cpp/pull/7/files (opened by someone else)
3. boost- https://github.com/starkware-libs/crypto-cpp/pull/13
4. googletest as dependecy that is fetched - https://github.com/starkware-libs/crypto-cpp/pull/14
5. Copying shared libs that are not .so files - https://github.com/starkware-libs/crypto-cpp/pull/15

**Pull request closed without merging**